system_tests/manager/test_hostpool_plugin.py

```diff
@@ -0,0 +1,188 @@
+
+import os
+
+from cosmo_tester.test_suites.test_blueprints import nodecellar_test
+from cosmo_tester.framework import util
+
+from system_tests import resources
+
+REPO_URL = 'https://github.com/cloudify-cosmo/cloudify-host-pool-plugin.git'
+
+
+os.environ['HANDLER_CONFIGURATION'] = '/home/elip/dev/system-tests-handlers/hp-openstack-region-b-dev2-handler.yaml'
+
+
+class HostPoolPluginTest(nodecellar_test.NodecellarAppTest):
+
+    def test_nodecellar_hostpool(self):
+        self._provision_pool()
+        self._install_host_pool_service()
+        self._test_nodecellar_impl(
+            'examples/nodecellar/host-pool-blueprint.yaml')
+        self._teardown_pool()
+        self._uninstall_host_pool_service()
+
+    def assert_monitoring_data_exists(self):
+        # this blueprint does not define monitoring
+        pass
+
+    def _provision_pool(self):
+
+        self.password_pool_host = util.generate_password()
+
+        def _render_password_authentication_script():
+
+            script_template = resources.get_resource(
+                'enable_password_authentication.sh.template')
+            return util.render_template(
+                template_path=script_template,
+                password=self.password_pool_host)
+
+        blueprint_path = resources.get_resource(
+            'pool-blueprint/pool-blueprint.yaml')
+        self.blueprint_yaml = blueprint_path
+
+        blueprint_id = '{0}-pool-blueprint'.format(self.test_id)
+        deployment_id = '{0}-pool-deployment'.format(self.test_id)
+
+        script = _render_password_authentication_script()
+
+        self.upload_deploy_and_execute_install(
+            blueprint_id=blueprint_id,
+            deployment_id=deployment_id,
+            inputs={
+                'image': self.env.ubuntu_image_id,
+                'flavor': self.env.small_flavor_id,
+                'enable_password_authentication_script': script
+            }
+        )
+        self.pool_deployment_id = deployment_id
+
+    def _teardown_pool(self):
+        self.execute_uninstall(self.pool_deployment_id)
+
+    def _install_host_pool_service(self):
+
+        def _render_pool():
+
+            hosts = self.client.deployments.outputs.get(
+                deployment_id=self.pool_deployment_id).outputs['hosts']
+
+            pool_template = resources.get_resource(
+                'host-pool-service-blueprint/pool.yaml.template')
+            util.render_template_to_file(
+                template_path=pool_template,
+                file_path=os.path.join(
+                    os.path.dirname(pool_template),
+                    'pool.yaml'),
+                ip_pool_host_1=hosts['host_1']['ip'],
+                ip_pool_host_2=hosts['host_2']['ip'],
+                public_address_pool_host_1=hosts['host_1']['public_address'],
+                public_address_pool_host_2=hosts['host_2']['public_address'],
+                password_pool_host=self.password_pool_host
+            )
```

Comment:
move host-pool service blueprint directory to temp and render template in that directory